solr/core/src/java/org/apache/solr/search/FunctionQParser.java

```diff
@@ -361,7 +361,9 @@ protected ValueSource parseValueSource(int flags) throws SyntaxError {
         ((FunctionQParser)subParser).setParseMultipleSources(true);
       }
       Query subQuery = subParser.getQuery();
-      if (subQuery instanceof FunctionQuery) {
+      if (subQuery == null) {
+        valueSource = new DoubleConstValueSource(0.0f);
+      } else if (subQuery instanceof FunctionQuery) {
         valueSource = ((FunctionQuery) subQuery).getValueSource();
       } else {
         valueSource = new QueryValueSource(subQuery, 0.0f);
```

Comment:
Should we add a test in QueryValueSource constructor to require non-null?

Reply:
I think it would be good to catch problems at the gate, adding this extra check. However, this could result in a risk that these exceptions pop up in places that were (seemed) executing fine. Very small risk, if a QueryValueSource with a null query is created a NPE is inevitable, so I cannot imagine adding such check will cause new problems. Still, I did not dare to add it yet. Can we get a veto of someone on this? (In the meantime I'll try it and run the test suite.)

Reply:
I pushed a commit adding the null check to the QueryValueSource constructor